src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1007,18 +1007,48 @@ DQLiftGatherSymCW::DQLiftGatherSymCW() {
         auto matched_out_ids = uat::_(node_to_output).at_or_at(cvtids, pids);
         const auto& matched_out_gather = node_to_output.at(gather);
 
+        // Handle case with non-f16 scale
+        if (matched_out_s.get_element_type() != ov::element::f16) {
+            auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, ov::element::f16);
+            auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
+            auto new_g_w = std::make_shared<ov::op::v8::Gather>(new_cvt_w, matched_out_ids, gather_c);
+            auto new_cvt_s = std::make_shared<ov::op::v0::Convert>(matched_out_s, ov::element::f16);
+            auto new_g_s = std::make_shared<ov::op::v8::Gather>(new_cvt_s, matched_out_ids, gather_c);
+            auto new_mul = std::make_shared<ov::op::v1::Multiply>(new_g_w, new_g_s);
+            auto new_out = std::make_shared<ov::op::v0::Convert>(new_mul, matched_out_s.get_element_type());
```

Comment:
What's the reason to nearly fully duplicate the flow here? why slightly changing the existing code isn't enough?

Reply:
Refactored